snakemake/scheduler.py

```diff
@@ -356,18 +362,20 @@ def schedule(self):
 
                 # handle errors
                 if user_kill or (not self.keepgoing and errors):
-                    if user_kill == "graceful":
+                    if (user_kill == "graceful" or self.keepgoing) and running:
                         logger.info(
                             "Will exit after finishing " "currently running jobs."
                         )
-
-                    if not running:
```

Comment:
previously, it would wait for running jobs, even when `keepgoing` is false, as `shutdown()` waits for all futures to complete